### PR TITLE
Reactivate golint in staging

### PR DESCRIPTION
**Pull request by sttts**
Run golint against staging with `staging/src/k8s.io/client-go/pkg` as the only exception. While doing this reset `hack/.linted_packages` which collected a lot of cruft. We now enforce that gone packages are removed.

### CODE REVIEW

File: hack/verify-golint.sh [hunk 1]
```diff
@@ -94,18 +100,29 @@ fi
 
 # check to make sure all packages that pass lint are in the linted file.
 echo
-if [ ${#linted[@]} -eq 0 ]; then
+if [ ${#linted[@]} -eq 0 -a ${#gone[@]} -eq 0 ]; then
```
Comment (deads2k) [inline hack/verify-golint.sh]: @stevekuznetsov does this do what it's supposed to do?  It doesn't have to be the pinnacle of beauty, it just needs to work.

**Comment by deads2k** [comment]
lgtm as long as someone who reads bash says that conditional stuff works.

File: hack/verify-golint.sh (hunk 1 above)
Comment (stevekuznetsov) [inline hack/verify-golint.sh]: You're running `#!/bin/bash` -- always use `[[` over `[`. If using `[[`, use `&&` and `||` for logic. If you must be POSIX compliant, use `[` and `-a` and/or `-o`. Access to empty arrays with `-o nounset` is tricky but `${#array[@]}` is safe.

File: hack/verify-golint.sh [hunk 2]
```diff
@@ -76,6 +76,12 @@ for p in "${all_packages[@]}"; do
 	fi
 done
 
+# Check that all linted_packages actually still exist
+gone=()
+for p in "${linted_packages[@]}"; do
+	array_contains "$p" "${all_packages[@]}" || gone+=( "$p" )
```
Comment (stevekuznetsov) [inline hack/verify-golint.sh]: More readable would be

```sh
if ! array_contains "${p}" "${all_packages[@]}"; then
    gone+=( "${p}" )
fi
```

**Comment by deads2k** [comment]
@stevekuznetsov said this should work (even if the spelling isn't stylistically pretty).  I've tagged it to stop the rotting.

**Comment by spxtr** [comment]
@k8s-bot test this

github api borked for a moment

**Comment by sttts** [comment]
@k8s-bot verify test this

**Comment by sttts** [comment]
@k8s-bot unit test this

**Comment by sttts** [comment]
Overriding approver label as this was lgtm'ed long before approvers were activated, caught in rebase hell getting through the queue.

**Pull request merged**